### PR TITLE
feat: handle invalid user tokens [ROAD-1059]

### PR DESCRIPTION
**Pull request by asaf92**
### Description

Sends a message when the user is not authenticated.

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### CODE REVIEW

File: infrastructure/cli/auth/initializer.go [hunk 1]
```diff
@@ -45,28 +44,43 @@ func NewInitializer(authenticator snyk.AuthenticationService, errorReporter erro
...
 
 	notification.Send(sglsp.ShowMessageParams{Type: sglsp.Info, Message: "Authenticating to Snyk. This could open a browser window."})
 
-	token, err := i.authenticator.Provider().Authenticate(context.Background())
+	token, err := authenticator.Provider().Authenticate(context.Background())
 	if token == "" || err != nil {
+		if err != nil {
+			err = &AuthenticationFailedError{}
+		}
 		log.Error().Err(err).Msg("Failed to authenticate. Terminating server.")
```
Comment (asaf92) [inline infrastructure/cli/auth/initializer.go:70]: Change the log message

File: domain/snyk/scanner.go [hunk 2]
```diff
@@ -69,7 +69,12 @@ func (sc *DelegatingConcurrentScanner) Scan(
 	done := make(chan bool)
 	defer close(done)
 
-	sc.initializer.Init()
+	err := sc.initializer.Init()
+	if err != nil {
+		log.Error().Err(err).Msg("Scan initialization error, cancelling scan")
```
Comment (bastiandoetsch) [inline domain/snyk/scanner.go:74]: would logging to Sentry make sense here?

**Review by bastiandoetsch — COMMENTED** [review]
looks good, we will merge later as talked about.

File: domain/snyk/scanner.go (hunk 2 above)
Comment (asaf92) [reply domain/snyk/scanner.go:74]: The initializers report to Sentry so I don't think there's a need

**Pull request merged**